src/Common/Config/YAMLParser.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <string>
+
+#include <Poco/DOM/Document.h>
+#include "Poco/DOM/AutoPtr.h"
+#include <yaml-cpp/yaml.h>
+#include <common/logger_useful.h>
+
+namespace DB
+{
+
+class YAMLParser
+{
+public:
+    YAMLParser();
+
+    /// loads yaml file into a YAML::Node
+    Poco::AutoPtr<Poco::XML::Document> parse(const String& path);
+
+    ~YAMLParser() = default;
+
+private:
+    /// creates an XML element with the same name and appends it to original_node's parent node
+    static Poco::AutoPtr<Poco::XML::Element> createCloneNode(Poco::XML::Element & original_node);
```

Comment:
It's possible to put this function and `processNode()` into an anonymous namespace in `YAMLParser.cpp`. And because of both of those functions have been written only to implement `parse()` (they're not related to the public interface of this class), it's better to do that.